client/mapRenderer/MapViewController.cpp

```diff
@@ -173,13 +216,15 @@ void MapViewController::onHeroMoved(const CGHeroInstance * obj, const int3 & fro
 	{
 		// instant movement
 		context->addObject(movingObject);
-		setViewCenter(movingObject->visitablePos());
+
+		if (actionVisible)
+			setViewCenter(movingObject->visitablePos());
 	}
 }
 
 void MapViewController::onHeroRotated(const CGHeroInstance * obj, const int3 & from, const int3 & dest)
 {
-	//TODO
+	//TODO. Or no-op?

```

Comment:
reminds me of https://youtu.be/vCtik1q_IsQ?t=13

Reply:
No such animations in H3 files unfortunately.

Reply:
What this method should do?

Reply:
Most likely - nothing. So will remove it once I confirm this.

This is for handling situation where hero need to face object to interact with. For example, hero is moving alongside road and wants to pick resource pile next to the road.
In this case there will be movement event/netpack but hero must only turn to face object (e.g. resource), without actual movement.

But since hero rotation is set by netpack applier and H3 has no animations for turning around there is no need to do anything in map rendering.